File: Data Structures & Algorithms/maximum-product-subarray/submission-1.go
func maxProduct(nums []int) int {
    res := nums[0]
    curMax, curMin := nums[0], nums[0]
    for i := 1; i < len(nums); i++ {
        num := nums[i]
        prevMax, prevMin := curMax, curMin
        curMax = max(num, max(prevMax*num, prevMin*num))
        curMin = min(num, min(prevMax*num, prevMin*num))
        res = max(res, curMax)
    }
    return res
}

File: Data Structures & Algorithms/insert-new-interval/submission-1.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []
        i = 0
        while i < len(intervals):
            begin, end = intervals[i][0], intervals[i][1]
            if begin > newInterval[1]:
                break;
            elif end < newInterval[0]:
                res.append(intervals[i])
                i += 1
            else:
                newInterval = [min(newInterval[0], begin), max(newInterval[1], end)]
                i += 1
            print(newInterval)
        res.append(newInterval)
        while i < len(intervals):
            res.append(intervals[i])
            i += 1
        return res